### PR TITLE
Conditions, Overview and issue feedback

### PR DESCRIPTION
**Pull request by cdvoisin**
This PR contains updates to:

1. Conditions: a new format based on proposals discussed. Based on "Glob" but that isn't a spec we can reference and is file-specific so I introduced detailed description within GA4GH Passport spec. Not sure how powerful we want the "pattern" system, and perhaps could drop the <open-bracket> feature and just go with <question-mark> by itself. Feedback please.

2. Overview updates of diagram and tweaks to the text that describes it.

3. Attempt to address some issues:
    a. Remove DURI from spec? #18
    b. Drop term attestation #22
    c. Undefined term Client #23
    d. Restrictive URL field definition #24
    e. Several definitions for registered access #25

### CODE REVIEW

File: researcher_ids/RI_Claims_V1.md [hunk 1]
```diff
@@ -585,106 +592,255 @@ the GA4GH DURI committee.
...
+    Object](#passport-visa-object) such that:
+
+    -   Checking the correctness of the Condition Clause MUST be performed first.
+        For example, a `type` field MUST be present.
+
+    -   Ignore Passport Visa Objects that have the `conditions` field present.
+        This will avoid deep nesting of condition evaluation (i.e. avoid condition
+        loops, stack overflows, etc).
+
+    -   A Condition Clause field matches when the `<match-type>` algorthm
```
Comment (martin-kuba) [inline researcher_ids/RI_Claims_V1.md]:
```suggestion
    -   A Condition Clause field matches when the `<match-type>` algorithm
```

File: researcher_ids/RI_Claims_V1.md [hunk 2]
```diff
@@ -585,106 +592,255 @@ the GA4GH DURI committee.
...
+        -   `source` = "https://visas.elixir.org"; AND
 
-    -   `value` = "faculty\@uni-heidelberg.de" AND `by` = "so"
+        -   `by` = "system"
 
-    -   `value` = "faculty\@uni-heidelberg.de" AND `by` = "system"
+##### Pattern Matching
 
-    -   `value` = "student\@uni-heidelberg.de" AND `by` = "so"
+-   MUST Use full string case-sensitive character pattern comparision.
```
Comment (martin-kuba) [inline researcher_ids/RI_Claims_V1.md]:
```suggestion
-   MUST Use full string case-sensitive character pattern comparison.
```

File: researcher_ids/RI_Claims_V1.md [hunk 3]
```diff
@@ -585,106 +592,255 @@ the GA4GH DURI committee.
...
+        by a colon and then string suffix, except for "type" where it MUST be
+        assumed to be "const" and is not specified.
+        
+        -   If prefix is "const", then suffix MUST use case sensitive full string
+            matching.
+
+        -   If prefix is "pattern", then suffix MUST use full string [Pattern
+            Matching](#pattern-matching) to match input.
+
+        -   If prefix is "split_pattern", then the full suffix MUST use full
```
Comment (martin-kuba) [inline researcher_ids/RI_Claims_V1.md:673]: I do not like that the split_pattern uses a single predefined character as the point for splitting. I would either remove the split_pattern altogether to simplify implementations, or allow it to specify arbitrary character for splitting. Just my two cents.

**Review by martin-kuba — APPROVED** [review]
Seems reasonable and in line with glob implementations in Java and Python.

File: researcher_ids/RI_Claims_V1.md [hunk 4]
```diff
@@ -237,38 +245,41 @@ for more details).
...
-between these components (represented by black arrows) MAY not be direct or
-conversely services shown as being separate MAY be combined into one service.
-For example, some implementations MAY deploy one service that handles the
-responsiblities of both the Passport Visa Issuer and the Passport Broker.
+Implementations may introduce clients, additional services, and protocols --
+not detailed in Diagram 2 -- to provide the mechanisms to move the data between
+the Passport Visa Assertion Source and the [Passport Broker](#passport-broker).
+These mechanisms are unspecified by the scope of this specification except that
+they MUST adhere to security and privacy best practices, such as those outlined
+in the GA4GH AAI Specification, in their handling of user sessions, Passport
```
Comment (mikael-linden) [inline researcher_ids/RI_Claims_V1.md]: I don't think AAI spec defines _a user session_ between the broker and assertion source. At least in ELIXIR/EGA we have none.

File: researcher_ids/RI_Claims_V1.md [hunk 5]
```diff
@@ -1194,16 +1355,24 @@ reader-friendly.
             "value": "https://ega-archive.org/datasets/00000432",
```
Comment (mikael-linden) [inline researcher_ids/RI_Claims_V1.md]:
```suggestion
            "value": "https://ega-archive.org/datasets/EGAD00000000001",
```
A real-world-example.

File: researcher_ids/RI_Claims_V1.md [hunk 6]
```diff
@@ -1194,16 +1355,24 @@ reader-friendly.
...
             "source": "https://grid.ac/institutes/grid.225360.0",
```
Comment (mikael-linden) [inline researcher_ids/RI_Claims_V1.md]:
```suggestion
            "source": "https://ega-archive.org/dacs/EGAC00001000205",
```
A DAC identifier for the dataset above.

Comment (mikael-linden) [reply researcher_ids/RI_Claims_V1.md]: The EGA service of EBI does not run any DACs. The DACs are nominated by the data owners. So the Passport Visa Assertion Source is the DAC although the Passport Visa Assertion Repository and Signatory are probably EBI.

**Review by mikael-linden — APPROVED** [review]
The changes look good.

File: researcher_ids/RI_Claims_V1.md (hunk 1 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: done

File: researcher_ids/RI_Claims_V1.md (hunk 2 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: done

File: researcher_ids/RI_Claims_V1.md (hunk 3 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md:673]: Adding extra text around this use case to precondition its use where a Passport Visa Type designs the splitting, such as it has been done for LinkedIdentities. We can look at a separate PR to introduce variables or other means to specify the separator, but as now written this capability is essentially saying that if you are planning on using multi-valued strings, use URI-encoded substrings with semi-colon separators like what LinkedIdentities does.

File: researcher_ids/RI_Claims_V1.md (hunk 4 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: Changed "user sessions" to "protocols", thanks for catching this.

File: researcher_ids/RI_Claims_V1.md (hunk 5 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: done

File: researcher_ids/RI_Claims_V1.md (hunk 6 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: Done

File: researcher_ids/RI_Claims_V1.md (hunk 5 above)
Comment (cdvoisin) [reply researcher_ids/RI_Claims_V1.md]: done

**Comment by cdvoisin** [comment]
Thanks for the review. We can follow up with last comments on "pattern" or "split_pattern" on another PR if needed.

**Pull request merged**